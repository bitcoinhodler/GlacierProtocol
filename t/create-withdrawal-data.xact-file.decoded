{
  "txid": "3e22466b22e958b6cccc5031736e14a768b0f2ac547c221be6090150a01ef1df",
  "hash": "cea3969a267efdf2cef00b96fcc38d30f4d8b7e2b2e2b6329419784948146ff1",
  "version": 2,
  "size": 374,
  "vsize": 184,
  "weight": 734,
  "locktime": 0,
  "vin": [
    {
      "txid": "c37257ccb80023105a67b47991c78fbb1889f394d9c29e8eb919c1c0b15f8554",
      "vout": 3,
      "scriptSig": {
        "asm": "0020d017965b0cae52a8b6f6e43b00118be27dae84b825959dcb77ca6b5310e53ea3",
        "hex": "220020d017965b0cae52a8b6f6e43b00118be27dae84b825959dcb77ca6b5310e53ea3"
      },
      "txinwitness": [
        "",
        "304402203ee99acd409742578a808c4f16eb61b998d795e43f6ba9aab5ada52b81e5b76002203c4e6dc1216881eb28898e13299fcd3789c4c1c668b81cd697af52e891c65cc901",
        "304402204ec63ece20d482500885570049fc3253fa2bdc4798e05b20794c9bbd9ec3114502207698df1405822fc4bf8fc6892949de4b5be3b3d0c360c48cfb21e17d2981a61f01",
        "5221035a0cf2b8ad46945154d80b339f730ac0cdbc39a95550a95821adf6df6e6e3c9421038f339e9149fda8496360d689b5d6b4d66f8e64e28b1c89846efd0831512eab882103442945263f31819baf5799dc9595eba49b8f6674dadf21189f717abd630ab15053ae"
      ],
      "sequence": 4294967295
    }
  ],
  "vout": [
    {
      "value": 0.19998160,
      "n": 0,
      "scriptPubKey": {
        "asm": "OP_DUP OP_HASH160 b6c7733e23a271e57ca16e5cd885174917968db9 OP_EQUALVERIFY OP_CHECKSIG",
        "desc": "addr(mxBQD1QAYpwiudaCJdRhE9QSW9cokafJ99)#fxf0ktmw",
        "hex": "76a914b6c7733e23a271e57ca16e5cd885174917968db988ac",
        "address": "mxBQD1QAYpwiudaCJdRhE9QSW9cokafJ99",
        "type": "pubkeyhash"
      }
    }
  ]
}
