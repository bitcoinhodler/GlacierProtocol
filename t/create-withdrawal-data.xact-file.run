#!/bin/bash
set -e

$GLACIERSCRIPT --regtest=$1 create-withdrawal-data << INPUT
y
y
y
y
y
y
2MvQKEWTdtH7uM5C72quuMukWNFkFYjm34N
5221035a0cf2b8ad46945154d80b339f730ac0cdbc39a95550a95821adf6df6e6e3c9421038f339e9149fda8496360d689b5d6b4d66f8e64e28b1c89846efd0831512eab882103442945263f31819baf5799dc9595eba49b8f6674dadf21189f717abd630ab15053ae
mxBQD1QAYpwiudaCJdRhE9QSW9cokafJ99
1
../../t/create-withdrawal-data.xact-file.tx
2
cMvAmArzxkXMh8k5FcaRWLBA2SgDSc2U8q1YE5hSLSek1GuyFBP3
cPSsBu9SyNVAS2Evy3m4ELFx7KGnudH3N77Es83nafa2xVWJGRSe
10
y

y
INPUT
