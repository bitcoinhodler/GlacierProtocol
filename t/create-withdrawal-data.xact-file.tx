0200000001cb573dd32763342cd37f3e76335273ecce71cd30eb69f61c2935bfeb1a7575bc010000006a47304402203907d6381cc51f3d2507bea94108803c651dd56940bc763cf4092d9ce753aa4502201d88660812c9f985283412eaecc99ee660bec3ad7ea1c2b9847ec2c1fbb293170121035c0aa8bd8dc73ea38dd76c5884d0a4308426cdd88913937912154fffc9955f56ffffffff05002d31010000000017a914b3d8e8424c22422fade911a92229e9095d18781e8790409700000000001976a9145472da4e4d09cfb0a96b6f0b973d49b67735f7b288ac002d31010000000017a914da56c1ede441385e70a0b7cfc79b11649eb2079087002d31010000000017a91422a07fe0ea8b8293eb336b9423f7e3958917924387002d31010000000017a9147a4f6388e8ea3dc6d1a3784bbfe214a46c7928998700000000
