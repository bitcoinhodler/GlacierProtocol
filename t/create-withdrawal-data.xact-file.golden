Are you running this on a computer WITHOUT a network connection of any kind? (y/n)?Have the wireless cards in this computer been physically removed? (y/n)?Are you running on battery power? (y/n)?Are you running on an operating system booted from a USB drive? (y/n)?Is your screen hidden from view of windows, cameras, and other people? (y/n)?Are smartphones and all other nearby devices turned off and in a Faraday bag? (y/n)?
You will need to enter several pieces of information to create a withdrawal transaction.


*** PLEASE BE SURE TO ENTER THE CORRECT DESTINATION ADDRESS ***


Source cold storage address: 
Redemption script for source cold storage address: 
Destination address: 
How many unspent transactions will you be using for this withdrawal? 
Please paste raw transaction #1 (hexadecimal format) with unspent outputs at the source address
OR
input a filename located in the current directory which contains the raw transaction data
(If the transaction data is over ~4000 characters long, you _must_ use a file.):

Transaction data found for source address.
TOTAL unspent amount for this raw transaction: 0.20000000 BTC

How many private keys will you be signing this transaction with (at least 2 required to complete transaction)?
#: Key #1: Key #2: 
Enter fee rate.
Satoshis per vbyte: 
Based on the provided rate, the fee will be 0.00002160 bitcoin.
Confirm? (y/n): 
Please enter the decimal amount (in bitcoin) to withdraw to the destination address.

Example: For 2.3 bitcoins, enter "2.3".

After a fee of 0.00002160, you have 0.19997840 bitcoins available to withdraw.

*** Technical note for experienced Bitcoin users:  If the withdrawal amount & fee are cumulatively less than the total amount of the unspent transactions, the remainder will be sent back to the same cold storage address as change. ***

Amount to send to mxBQD1QAYpwiudaCJdRhE9QSW9cokafJ99 (leave blank to withdraw all funds stored in these unspent transactions): With no change output, the transaction fee is reduced, and 0.19998160 BTC will be sent to your destination.

Is this data correct?
*** WARNING: Incorrect data may lead to loss of funds ***

0.20000000 BTC in unspent inputs from cold storage address 2MvQKEWTdtH7uM5C72quuMukWNFkFYjm34N
0.19998160 BTC going to destination address mxBQD1QAYpwiudaCJdRhE9QSW9cokafJ99
Fee amount: 0.00001840

Signing with private keys: 
cMvAmArzxkXMh8k5FcaRWLBA2SgDSc2U8q1YE5hSLSek1GuyFBP3
cPSsBu9SyNVAS2Evy3m4ELFx7KGnudH3N77Es83nafa2xVWJGRSe


Confirm? (y/n): 
Calculating transaction...

Final fee rate: 10.00 satoshis per vbyte

Raw signed transaction (hex):
0200000000010154855fb1c0c119b98e9ec2d994f38918bb8fc79179b4675a102300b8cc5772c30300000023220020d017965b0cae52a8b6f6e43b00118be27dae84b825959dcb77ca6b5310e53ea3ffffffff01d0253101000000001976a914b6c7733e23a271e57ca16e5cd885174917968db988ac040047304402203ee99acd409742578a808c4f16eb61b998d795e43f6ba9aab5ada52b81e5b76002203c4e6dc1216881eb28898e13299fcd3789c4c1c668b81cd697af52e891c65cc90147304402204ec63ece20d482500885570049fc3253fa2bdc4798e05b20794c9bbd9ec3114502207698df1405822fc4bf8fc6892949de4b5be3b3d0c360c48cfb21e17d2981a61f01695221035a0cf2b8ad46945154d80b339f730ac0cdbc39a95550a95821adf6df6e6e3c9421038f339e9149fda8496360d689b5d6b4d66f8e64e28b1c89846efd0831512eab882103442945263f31819baf5799dc9595eba49b8f6674dadf21189f717abd630ab15053ae00000000

Transaction fingerprint (md5):
13e96f146d77029cf2d6a268668c1e09
QR code for transaction written to transaction.png
